moped-editor/src/views/projects/projectView/ProjectComponents/CTNSourcesAndLayers.js

```diff
@@ -0,0 +1,93 @@
+import { Source, Layer } from "react-map-gl";
+import { SOURCES } from "./mapSettings";
+import { MAP_STYLES } from "./mapStyleSettings";
+
+/**
+ * Component that renders CTN GeoJSON data to be captured into draft or edited components
+ * All layers are set to show below basemap street labels using beforeId = "street-labels"
+ * @param {Boolean} isCreatingComponent - are we creating a component?
+ * @param {Boolean} isEditingComponent - are we editing a component?
+ * @param {String} linkMode - Tracks if we are editing "lines" or "points"
+ * @param {Boolean} isDrawing - is the user currently drawing?
+ * @param {Object} ctnLinesGeojson - GeoJSON data for CTN lines in the visible area of the map
+ * @param {Object} ctnPointsGeojson - GeoJSON data for CTN points in the visible area of the map
+ * @returns JSX.Element
+ */
+const CTNSourcesAndLayers = ({
+  isCreatingComponent,
+  isEditingComponent,
+  linkMode,
+  isDrawing,
+  ctnLinesGeojson,
+  ctnPointsGeojson,
+}) => {
+  const isEditingLines =
+    (isCreatingComponent || isEditingComponent) &&
+    linkMode === "lines" &&
+    !isDrawing;
+  const isEditingPoints =
+    (isCreatingComponent || isEditingComponent) &&
+    linkMode === "points" &&
+    !isDrawing;
```

Comment:
This is the important change here - we show the dashed CTN lines or grey intersection dots in the map UI when we are creating or editing a component.